File: 104-maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func maxDepth(root *TreeNode) int {
    // If root is null, return 0
    if root == nil {
        return 0
    }

    // Otherwise, return max between recursive left/right calls + 1
    return max(maxDepth(root.Left), maxDepth(root.Right)) + 1
}

File: 104-maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int maxDepth(TreeNode root) {
        // If root is null, return 0
        if (root == null) return 0;

        // Otherwise, return max between recursive left/right calls + 1
        return Math.max(maxDepth(root.left), maxDepth(root.right)) + 1;
    }
}